ops/docker/Dockerfile.packages

```diff
@@ -1,10 +1,37 @@
 # This Dockerfile builds all the dependencies needed by the monorepo, and should
 # be used to build any of the follow-on services
 #
-# ### BASE: Install deps
+
+# Stage 0 (named `manifests`) collects
+# dependency manifest files (`package.json` and `yarn.lock`) which are then
+# used by stage 1 to install these dependencies
+# development. The only reason we need a separate stage just for collecting the
+# dependency manifests is that Docker's `COPY` command still does not allow
+# copying based on a glob pattern (see this GitHub issue for more details
+# https://github.com/moby/moby/issues/15858). Being able to copy only manifests
+# into stage 1 (the `COPY --from=manifests` statement) is important to maximize
+# Docker build cache hit rate. `alpine` is chosen as the base image for the
+# first stage because it's the smallest image that have access to the `cp
+# --parents -t` command (by installing the `coreutils` package).
+FROM alpine:3.16 as manifests
+RUN apk add coreutils
```

Comment:
When/if we switch to pnpm pnpm has a really nice way of accomplishing this where you install the node modules directly from only lockfile and not package.jsons